src/file.js

```diff
@@ -153,6 +153,25 @@
   };
 
   /**
+   * Method to replace an extension, if one does not exist in the file string, it will be added.
+   *
+   * @param {string} fileString - File string to add the extension to if one does not exist
+   * @param {string} extension - Extension to add if one does not exist in `fileString`. The
+   *   value is the extension without the `.`. E.g. `js`, `html`.  Not `.js`, `.html`.
+   * @returns {string} New fileString with the new extension if one did not exist
+   */
+  File.replaceExtension = function(fileString, extension) {
+    var fileName  = File.parseFileName(fileString),
```

Comment:
Unfortunately, even though I like the idea of using a simple regex, this will break in the following case: `a/.file` (which could be seen as `directory/.bashrc`, for example). `bashrc` in the previous example isn't really the extension, it is the filename.

`directory/.test/test` will also break as `.test/test` will be selected as the extension. I'm not really sure where to go from here, other than creating a complex if/then/else statement.